src/plugins/legacy/polygonRoi/readers/medContoursReader.cpp

```diff
@@ -0,0 +1,374 @@
+        {
+            return readBinaryFile(path);
+        }
+        else
+        {
+            // TODO ==> Read ASCII File to implement
+            return false;//readASCIIFile(path);
+        }
+    }
+
+    return false;
+}
+
+bool medContoursReader::readBinaryFile(QString path)
+{
+    QFile file(path);
+    if (file.open(QFile::ReadOnly))
+    {
+        if (d->fileVersion == medContoursWriter::EXPORT_VERSION_1)
+        {
+            QDataStream in(&file);
+
+            qint64 dummy = 0, startMeshPos = 0, endMeshPos = 0, fileEndPos;
+            // skip header
+            in >> dummy >> startMeshPos >> endMeshPos >> fileEndPos;
+
+            // the VTK mesh part starts here. Determine size of the
+            // mesh block
+            qint64 meshLen = endMeshPos - startMeshPos;
+
+            // hopefully this cast goes well...
+            int len = static_cast<int>(meshLen);
+            char* inputString = new char[len];
+            if (in.readRawData(inputString, len) == -1)
+            {
+                qDebug() << metaObject()->className() << ":: read - failed to read binary mesh data.";
+                delete[] inputString;
+                file.close();
+                return false;
+            }
+            setProgress(25);
+
+            vtkSmartPointer<vtkDataSetReader> reader = vtkSmartPointer<vtkDataSetReader>::New();
+            reader->SetReadFromInputString(1);
+            reader->SetBinaryInputString(inputString, len);
+
+            vtkSmartPointer<vtkMetaDataSet> dataSet = retrieveMeshFromReader(reader);
+            if (!dataSet)
+            {
+                qDebug() << metaObject()->className() << ":: readBinaryFile - failure to read Contours.";
+                return false;
+            }
+            setProgress(60);
+
+            // retrieve contours with label
+            qint32 *size = new qint32();
+            in >> *size;
+            QVector<medTagContours> contoursSet;
+
+            for (int i=0; i<*size; i++)
+            {
+                medTagContours contours;
+                in >> contours;
+                contoursSet.append(contours);
+            }
+
+            setProgress(90);
+
+            if (file.pos() != fileEndPos)
+            {
+                qDebug() << metaObject()->className() << ":: wrong pos :" <<
+                            file.pos() << "instead of" << fileEndPos;
+            }
+
+            // set the mesh then the EP data
+            this->data()->setData(dataSet);
+            this->data()->setData(&contoursSet, 1);
+
+            file.close();
+        }
+        else
+        {
+            qDebug() << metaObject()->className() << ":: unsupported binary file.";
+            file.close();
+            return false;
+        }
+    }
+    setProgress(100);
+    return true;
+}
+
+bool medContoursReader::readASCIIFile(QString path)
+{
+//    vtkSmartPointer<vtkDataSetReader> reader = vtkSmartPointer<vtkDataSetReader>::New();
```

Comment:
code commenter

Reply:
Ok I remove this code but 
In writer, I have implemented the ability to write an ASCII file (.ctr)
For now, The only format supported is .ctrb (binary). 
We have just to implement a reader for ASCII file, then add a new supported format in writer